main.go

```diff
@@ -0,0 +1,422 @@
+
+	databaseClusters, err := client.DescribeDBClusters(&rds.DescribeDBClustersInput{DBClusterIdentifier: &d.DBClusterIdentifier})
+	if err != nil {
+		return errors.Wrap(err, "unable to describe the DB Cluster")
+	}
+
+	if len(databaseClusters.DBClusters) == 0 {
+		return errors.Wrap(err, "list of DB Clusters empty")
+	}
+	for _, member := range databaseClusters.DBClusters[0].DBClusterMembers {
+		if *member.DBInstanceIdentifier == d.DBInstanceIdentifier {
+			(*d).IsClusterWriter = *member.IsClusterWriter
+		}
+	}
+	return nil
+}
+
+func (d *DBInstance) getNewClassType() (string, error) {
+	newClass, err := d.increaseSize()
+	if err != nil {
+		return "", err
+	}
+	log.Infof("New DB instance class (%s)", newClass)
+	return newClass, nil
+}
+
+func (d *DBInstance) getSetDBInstanceClass() bool {
+	for i, dbClass := range DBInstanceClasses {
+		if d.DBInstanceClass == dbClass {
+			(*d).SizeIndex = i
+			return true
+		}
+	}
+	return false
+}
+
+func (d DBInstance) increaseSize() (string, error) {
+	newIndex := d.SizeIndex + 1
+	if (newIndex + 1) >= len(DBInstanceClasses) {
+		return "", errors.Errorf("Maximum instance size used. Index out of range")
+	}
+	return DBInstanceClasses[newIndex], nil
+}
+
+func (d *DBInstance) changeDatabaseClass(client *rds.RDS, dbInstanceClass string) error {
+	modifyDBInstanceInput := &rds.ModifyDBInstanceInput{
+		ApplyImmediately:     aws.Bool(true),
+		DBInstanceClass:      aws.String(dbInstanceClass),
+		DBInstanceIdentifier: aws.String(d.DBInstanceIdentifier),
+	}
+	log.Infof("Upgrading database (%s) to class (%s)", d.DBInstanceIdentifier, dbInstanceClass)
+	_, err := client.ModifyDBInstance(modifyDBInstanceInput)
+	if err != nil {
+		return errors.Wrap(err, "unable to upgrade database to new class")
+	}
+	wait := 300
+	log.Infof("Waiting up to %d seconds for db instance to start modifications...", wait)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	err = d.waitForDBInstanceStartModifications(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	wait = 1000
+	log.Infof("Waiting up to %d seconds for db instance to become available...", wait)
+	ctx, cancel = context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	err = d.waitForDBInstanceReady(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *DBInstance) waitForDBInstanceReady(ctx context.Context, client *rds.RDS) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "timed out waiting for database to get ready")
+		default:
+			var shouldWait bool
+
+			databaseInstances, err := client.DescribeDBInstances(&rds.DescribeDBInstancesInput{DBInstanceIdentifier: &d.DBInstanceIdentifier})
+			if err != nil {
+				log.WithError(err).Error("unable to describe DB instance")
+			}
+
+			if len(databaseInstances.DBInstances) == 0 {
+				log.Error("List of DB instances empty")
```

Comment:
You will want to do something like this, otherwise you will continue on and could get a panic when referencing `DBInstances[0]`.
```suggestion
				log.Error("List of DB instances empty")
				time.Sleep(5 * time.Second)
				continue
```

All of the errors that are logged in these wait loops should look something like this.